src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -515,93 +511,208 @@ void ConfigManager::parseConfig(const nlohmann::json& j) {
+    collect the ones, who only check one option, and assign those to that
+    option, in that order.
+    Then we collect all `ConfigManager`, including this instance, and assign
+    their held validators to them in the same order, as they saved them, unless
+    a validator was already assigned, in which case we skip it. `ConfigManager`
+    always add a newly created validator to the end of a vector, so they should
+    already be ordered by creation, if we only look at the validators in one
+    `ConfigManager`.
     */
-    const nlohmann::ordered_json::json_pointer jsonOptionPointer{path};
-
-    if (printCurrentJsonConfiguration) {
-      // We can only use the value, if we are sure, that the value was
-      // initialized.
-      configuratioOptionsVisualization[jsonOptionPointer] =
-          option.wasSet() ? option.getValueAsJson()
-                          : "value was never initialized";
-    } else {
-      configuratioOptionsVisualization[jsonOptionPointer] =
-          option.hasDefaultValue() ? option.getDefaultValueAsJson()
-                                   : option.getDummyValueAsJson();
-    }
+    ConfigurationDocValidatorAssignment assignment{};
+
+    // Assign to the configuration options.
+    const auto& allValidators = validators(true);
+    std::ranges::for_each(
+        std::views::filter(allValidators,
+                           [](const ConfigOptionValidatorManager& val) {
+                             return val.configOptionToBeChecked().size() == 1;
+                           }),
+        [&assignment](const ConfigOptionValidatorManager& val) {
+          // The validator manager only has one element, so this should be okay.
+          const ConfigOption& opt = **val.configOptionToBeChecked().begin();
+          assignment.addEntryUnderKey(opt, val);
+        });
+
+    // Assign to the configuration managers.
+    std::vector<std::reference_wrapper<const ConfigManager>> allManager{
+        ad_utility::transform(
+            allHashMapEntries("",
+                              [](const HashMapEntry& entry) {
+                                return entry.holdsSubManager();
+                              }),
+            [](const auto& pair) {
+              return std::reference_wrapper<const ConfigManager>(
+                  *pair.second.getSubManager().value());
+            })};
+    allManager.emplace_back(*this);
+    std::ranges::for_each(
+        allManager, [&assignment](const ConfigManager& manager) {
+          std::ranges::for_each(
+              std::views::filter(manager.validators_,
+                                 [&assignment](const auto& validator) {
+                                   return !assignment.containsValue(validator);
+                                 }),
+              [&assignment, &manager](const auto& validator) {
+                assignment.addEntryUnderKey(manager, validator);
+              });
+        });
```

Comment:
And extract this into a function `getValidatorAssignment`

Reply:
Done, see [b0bb18b](https://github.com/ad-freiburg/qlever/pull/1169/commits/b0bb18b65d4e3bc7ab495a6882798bd14f783976).